tests/src/Context/MinkContext.php

```diff
@@ -86,4 +89,108 @@ public function iWaitForAjaxToFinish($event = NULL) {
     parent::iWaitForAjaxToFinish($event);
   }
 
+  /**
+   * {@inheritdoc}
+   *
+   * Overrides the parent method in order to support Select2.
+   */
+  public function selectOption($select, $option): void {
+    if ($field = $this->select2IsUsed($select, $option)) {
+      $xpath = '//li[contains(@class, "select2-selection__choice")]/span[contains(@class, "select2-selection__choice__remove")]';
+      if ($selected_remove_buttons = $field->getParent()->findAll('xpath', $xpath)) {
+        // This method selects anew an option in a normal select list. To have
+        // the same results in a select2 widget, all selected options have to
+        // be removed manually.
+        // To select an additional item, use the ::additionallySelectOption
+        // method.
+        foreach ($selected_remove_buttons as $selected_remove_button) {
+          $selected_remove_button->click();
+        }
+      }
+
+      $this->selectSelect2Option($field, $option);
+      return;
+    }
+    parent::selectOption($select, $option);
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * Overrides the parent method in order to support Select2.
+   */
+  public function additionallySelectOption($select, $option): void {
+    if ($field = $this->select2IsUsed($select, $option)) {
+      $this->selectSelect2Option($field, $option);
+      return;
+    }
+    parent::additionallySelectOption($select, $option);
+  }
+
+  /**
+   * Checks if a given select field is using Select2.
+   *
+   * @param string $select
+   *   The select.
+   * @param string $option
+   *   The option to be selected.
+   *
+   * @return \Behat\Mink\Element\NodeElement|false
+   *   It returns the field as node element, if Select2 is used it or FALSE
```

Comment:
Typo. My mistake: "...if Select2 is used it or FALSE" should be "...if Select2 is use or FALSE"